libs/community/langchain_community/llms/bedrock.py

```diff
@@ -77,6 +77,22 @@ def _human_assistant_format(input_text: str) -> str:
     return input_text
 
 
+def _generate_messages_format(input_text: str) -> list:
```

Comment:
What about multiple turns for word in mouth of assistant turn?

It might be best to convert the old prompt formatting to the new messages by adding some logic to break things up Assistant Human turns into messages. This will give backwards compatibility for Claude 2 text completion prompts.

Reply:
The idea would be for Cluade 2 prompts such as,
 "prompt": "\n\nHuman: Write me a play. \n\nAssistant: Here is a play about cats, " 

would be translated to,
```
    "messages": [
        {
            "content": [
                {
                    "text": "Write me a play.",
                    "type": "text"
                }
            ],
            "role": "user"
        },
        {
            "content": [
                {
                    "text": "Sure, here is a play about cats,",
                    "type": "text"
                }
            ],
            "role": "assistant"
        }
    ]
```
By using the regex, this function could offer full breadth backwards compatibility to folks migrating their prompts from Claude 2, to Claude 3 models without having to migrate completion string prompts to Langchain messages.

When it comes to BedrockChat, I have an implementation on hand, let me see if I can merge that with your branch...

Reply:
The key thing here is that Bedrock itself has the model_id and input format "hardcoded" to expect certain inputs. 

That regex approach would work if Bedrock wasn't validating the input format -- we almost need to un-regex it and put it into the format that Bedrock expects for each Claude model. 

and then the trouble there is each Claude model has different input/output structures that you can't sniff out without differentiating as a "new" provider, `anthropic_v3`. 

the parameters for `prepare_input_stream` and `prepare_output_stream` has no model information in it, only provider -- so we can't programmatically say "for claude v2, do this kind of regex operation but for claude v3, leave it alone". 

to the LLMInputOutputAdapter class, it all just looks like "anthropic" without this new v3 moniker.

Reply:
The messages API also supports `system` property for a system prompt, so it can be more sophisticated than just Human/Assistant sequence. 

While playing with it, I found that Claude 3 and Claude 2 both work fine with messages API (i.e. no errors, and reasonable response), when you feed them a message that includes Human/Assistant sequence. I.e. w/o having a need to have a separate anthropic_v3 provider, you can feed the exact same `"prompt": "\n\nHuman: Write me a play. \n\nAssistant: Here is a play about cats, "` 

I made a change locally that looks like this
```
      if provider == "anthropic":
            if input_body.get("use_completion_api") is True:
                input_body["prompt"] = _human_assistant_format(prompt)
                if "max_tokens_to_sample" not in input_body:
                    input_body["max_tokens_to_sample"] = 256
                del input_body["use_completion_api"]
            else:
                input_body["messages"] = _generate_messages_format(prompt)
```   

And for Claude 2 it returns very similar results for the same prompt in both cases, when  `use_completion_api` is set, and when it's not. 

Though, to use messages API with Claude  2, `model_kwargs` need to be something like this:
```
model_kwargs_claude = {
                "temperature": 0,
                "top_k": 10,
                "max_tokens": 4000,
                "anthropic_version": "bedrock-2023-05-31",
            }
 ```